### PR TITLE
Fix#2021 Enhance SqlScriptLexer/SqlScriptParser to correctly deal with BEGIN…END blocks in sql scripts

### PR DESCRIPTION
**Pull request by sman-81**
This PR fixes issue #2021 (JDBI incorrectly splits statements for Oracle SQL script) by enhancing classes `SqlScriptLexer` and `SqlScriptParser` to correctly deal with `BEGIN…END` blocks in sql scripts.

It also adds a JUnit test (`testOracleScriptWithBeginEndBlock`) based on the failing script received from the original bug reporter (amended for better coverage).

### CODE REVIEW

**Comment by sman-81** [comment]
Hi there, could one of the maintainers kindly take a look at this PR and approve it?

File: core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java [hunk 1]
```diff
@@ -37,26 +47,37 @@ public String parse(CharStream charStream) {
...
-                case SqlScriptLexer.COMMENT:
-                case SqlScriptLexer.MULTI_LINE_COMMENT:
+                case COMMENT:
+                case MULTI_LINE_COMMENT:
                     break;
-                case SqlScriptLexer.NEWLINES:
+                case NEWLINES:
                     if (sb.length() > 0) {
-                        sb.append(' ');
+                        sb.append(System.lineSeparator()); // normalize newline sequences
```
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java:75]: Hm. This is a bit odd - if you run a Jdbi client program on Windows, suddenly your Postgres server on Linux will see Windows line endings. Different text will be sent to the server depending on the client OS which I find a bit surprising.

Is it harmful to keep the existing `' '`, or we could use a fixed `'\n'`?

File: core/src/test/java/org/jdbi/v3/core/statement/TestScript.java [hunk 2]
```diff
@@ -98,9 +101,44 @@ public void testScriptAsSetOfSeparateStatements() {
...
+
+            assertThat(statements).hasSize(3);
+
+            String lastStmt = statements.get(2);
+            assertThat(lastStmt).startsWith("CREATE OR REPLACE TRIGGER EXAMPLE_TRIGGER");
+            assertThat(lastStmt).endsWith("END;");
+            assertThat(lastStmt).hasLineCount(15);
+            assertThat(lastStmt).has(new Condition<>(s -> 7 == s.chars().filter(ch -> ch == ';').count(), "count semicolons"));
+
+        } catch (SQLException ex) {
```
Comment (stevenschlansker) [inline core/src/test/java/org/jdbi/v3/core/statement/TestScript.java:136]: You could just make the test `throws SQLException` which will automatically fail the test.

File: core/src/test/resources/script/oracle-with-begin-end-blocks.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+CREATE SEQUENCE EXAMPLE_ID_SEQUENCE
+    START WITH 1
+    INCREMENT BY 1;
+
+CREATE OR REPLACE TRIGGER EXAMPLE_TRIGGER
+    BEFORE INSERT
+    ON EXAMPLE
+    FOR EACH ROW
+BEGIN
```
Comment (stevenschlansker) [inline core/src/test/resources/script/oracle-with-begin-end-blocks.sql:30]: Would it be worth testing that `BEGIN` and `END` are ok to include in comments?
like `-- a comment that includes BEGIN`

**Comment by stevenschlansker** [comment]
Sorry for the delay @sman-81 , I was on vacation 🚢 
I think the test failure is a false positive. I left some comments for consideration.

File: core/src/test/java/org/jdbi/v3/core/statement/TestScript.java (hunk 2 above)
Comment (sman-81) [reply core/src/test/java/org/jdbi/v3/core/statement/TestScript.java:136]: Ok, will do

File: core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java (hunk 1 above)
Comment (sman-81) [reply core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java:75]: Database servers are newline-agnostic. They must be, dealing with clients of all sorts and os.
I didn't like the super-long lines the original code creates. But sure, I can roll this change back.

File: core/src/test/resources/script/oracle-with-begin-end-blocks.sql (hunk 3 above)
Comment (sman-81) [reply core/src/test/resources/script/oracle-with-begin-end-blocks.sql:30]: I've amended the test but please be aware I am anything but an Antlr expert ;)

**Comment by sman-81** [comment]
> Sorry for the delay @sman-81 , I was on vacation

Thanks for the feedback @stevenschlansker and welcome back from a well-deserved vacation ;)
you know PR authors can be eager beavers 😂
I have incorporated your recommendations and force-pushed the update.

**Review by stevenschlansker — APPROVED** [review]

**Pull request merged**